metaflow/plugins/kfp/tests/flows/resources_flow.py

```diff
@@ -91,8 +100,30 @@ def get_env_vars(env_resources: Dict[str, str]) -> List[V1EnvVar]:
         )
     )
 
+sub_dict = dict(x=1, y="hello")
+default_dict = dict(a=1, hello="world", sub=sub_dict)
+
+class TestTypeClass(click.ParamType):
```

Comment:
Please explain what this class is for. Also `JSONType` is imported but not used.

Reply:
because `JSONType` is specially handled and converted the value dumped to json here https://github.com/zillow/metaflow/blob/40c2c1074aa014d7c226e499ba0edc887ef5bda1/metaflow/plugins/kfp/kfp_cli.py#L250-L252

so I introduced `TestTypeClass` to ensure that JSON serialization would work for this `DatasetType` here https://github.com/zillow/datasets/blob/main/datasets/__init__.py#L13